sql/kvstore/nipost_challenge.go

```diff
@@ -0,0 +1,67 @@
+package kvstore
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+const NIPostChallengeKey = "NIPost"
+
+// AddNIPostChallenge adds the data for nipost to the key-value store.
```

Comment:
The logic in `AddNiPostChallenge()` is 100% the same as the logic in `AddNIPostBuilderState()`. The only difference is the type (`NiPostChallenge` vs `NIPostBuilderState`) and the key. The same probably applies to the `GetX()` functions. How about extracting the core DB logic to two separate functions:

```go
func Set(db sql.Executor, key []byte, value codec.Encodable) error;
func Get(db sql.Executor, key []byte, value codec.Decodable) error;
```

or like:

```go
func Get[T codec.Decodable](db sql.Executor, key []byte) (T, error);
```

Reply:
Pulled out shared logic and moved it to `kvstore.go`